sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/_io/base.py

```diff
@@ -69,11 +69,21 @@ def __init__(self, meta: Union[Input, Output], data, **kwargs):
         :type data: Union[None, int, bool, float, str
                           azure.ai.ml.Input,
                           azure.ai.ml.Output]
+        :param original_data: default value of input/output, None means un-configured data.
+        :type original_data: Union[None, int, bool, float, str
+                          azure.ai.ml.Input,
+                          azure.ai.ml.Output]
         """
         self._meta = meta
+        self._original_data = data
         self._data = self._build_data(data)
+        self._default_data = default_data
         self._type = meta.type if meta else kwargs.pop("type", None)
-        self._mode = self._data.mode if self._data and hasattr(self._data, "mode") else kwargs.pop("mode", None)
+        # pipeline level inputs won't pass mode to bound node level inputs
+        if isinstance(self._original_data, PipelineInput):
```

Comment:
what will self._data be if original data is PipelineInput? What's the difference between 
`self._mode = None`
and
`self._mode = self._data.mode ...`

Reply:
self._data will also be PipelineInput in this case.
So `self._mode = self._data.mode` will store PipelineInput's mode, which we don't want in node level

Reply:
then maybe extract it as a function and give some more information about different conditions? data can be PipelineInput, NodeOutput, Input, primitive values?